editors/code/src/config.ts

```diff
@@ -173,6 +174,24 @@ export class Config {
             requireReloadMessage = 'Changing cargo features requires a reload';
         }
         this.prevCargoFeatures = { ...this.cargoFeatures };
+        
+        if (
+            this.prevCargoWatchOptions !== null &&
+            (this.cargoWatchOptions.enable !==
+                this.prevCargoWatchOptions.enable ||
+                this.cargoWatchOptions.command !==
+                this.prevCargoWatchOptions.command ||
+                this.cargoWatchOptions.allTargets !==
+                this.prevCargoWatchOptions.allTargets ||
+                this.cargoWatchOptions.arguments.length !==
+                this.prevCargoWatchOptions.arguments.length ||
+                this.cargoWatchOptions.arguments.some(
+                    (v, i) => v !== this.prevCargoWatchOptions!.arguments[i],
+                ))
```

Comment:
the `if` case is a little bit large, I will prefer to use bool flags here .e.g:
```typescript
var changed = this.cargoWatchOptions.enable !== this.prevCargoWatchOptions.enable;
changed ||=  this.cargoWatchOptions.command !== this.prevCargoWatchOptions.command;
...
```

Reply:
I agree it's not great. `||=` doesn't appear to be an operator? and ts doesn't permit a bitwise version on bools either.  Regardless, I'll push a change that is in the vein of your nit and slightly less stacked ^^